// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;

option java_outer_classname = "RobotIdServiceProto";

import "bosdyn/api/robot_id.proto";

// RobotIdService provides mostly static identifying information about a robot.
// User authentication is not required to access RobotIdService to assist with
// early robot discovery.
service RobotIdService {
    // Get the robot id information. The ID contains basic information about a robot
    // which is made available over the network as part of robot discovery without
    //requiring user authentication.
    rpc GetRobotId(RobotIdRequest) returns (RobotIdResponse) {
    }
}

